src/vivarium_census_prl_synth_pop/components/businesses.py

```diff
@@ -272,46 +281,61 @@ def generate_businesses(self, pop_data: SimulantData) -> pd.DataFrame():
             }
         )
 
+        # Generate random employers
         pct_adults_needing_employers = 1 - known_employers["prevalence"].sum()
-        n_need_employers = np.round(n_working_age * pct_adults_needing_employers)
-        employee_counts = np.random.lognormal(
-            4, 1, size=int(n_need_employers // data_values.EXPECTED_EMPLOYEES_PER_BUSINESS)
+        n_need_employers = np.round(
+            self.us_population_size
+            * data_values.PROPORTION_18_PLUS
+            * pct_adults_needing_employers
+        )
+        n_businesses = int(n_need_employers // data_values.EXPECTED_EMPLOYEES_PER_BUSINESS)
+        random_employers_idx = np.arange(2, n_businesses + 2)
+        employee_count_propensity = self.business_moves_randomness.get_draw(
+            random_employers_idx
+        )
+        employee_counts = stats.lognorm.ppf(
+            q=employee_count_propensity, s=1, scale=np.exp(4)
         ).round()
-        n_businesses = len(employee_counts)
         random_employers = pd.DataFrame(
             {
-                "employer_id": np.arange(2, n_businesses + 2),
+                "employer_id": random_employers_idx,
                 "employer_name": ["not implemented"] * n_businesses,
-                "prevalence": employee_counts
-                / employee_counts.sum()
-                * pct_adults_needing_employers,
+                "prevalence": employee_counts / employee_counts.sum(),
                 "employer_address_id": np.arange(
                     len(known_employers), n_businesses + len(known_employers)
                 ),
             }
         )
 
+        # Combine and add details
         businesses = pd.concat([known_employers, random_employers], ignore_index=True)
-        # Randomly assign employer states/PUMAs
         states_pumas = randomly_sample_states_pumas(
             unit_ids=businesses.index,
             state_puma_options=self.state_puma_options,
             additional_key="initial_business_states_pumas",
-            randomness_stream=self.randomness,
+            randomness_stream=self.business_moves_randomness,
         )
         businesses["employer_state_id"] = states_pumas["state_id"]
         businesses["employer_puma"] = states_pumas["puma"]
-        self.employer_address_id_count = len(businesses)  # So next address will be unique
+
+        # Update employer_address_id_count
+        self.employer_address_id_count = len(businesses)
 
         return businesses.set_index("employer_id")
 
     def assign_random_employer(
-        self, sim_index: pd.Index, additional_seed: Any = None
+        self,
+        sim_index: pd.Index,
+        additional_seed: Any = None,
+        randomness_stream: RandomnessStream = None,
```

Comment:
We should always be using `self.randomness_stream` here, so you should remove the `randomness_stream` argument.